common/src/main/java/com/wynntils/features/inventory/ItemFavoriteFeature.java

```diff
@@ -121,4 +137,22 @@ private static void renderFavoriteItem(SlotRenderEvent.Post event) {
                 Texture.FAVORITE_ICON.width(),
                 Texture.FAVORITE_ICON.height());
     }
+
+    private void tryChangeFavoriteStateOnHoveredSlot(Slot hoveredSlot) {
+        if (hoveredSlot == null || !hoveredSlot.hasItem()) return;
+
+        String itemName = null;
+
+        Optional<NamedItemProperty> namedItemPropertyOpt =
+                Models.Item.asWynnItemProperty(hoveredSlot.getItem(), NamedItemProperty.class);
+        if (namedItemPropertyOpt.isPresent()) {
+            itemName = namedItemPropertyOpt.get().getName();
+        } else if (allowFavoritingAllItems.get()) {
+            itemName = normalizeBadString(
+                    StyledText.fromComponent((hoveredSlot.getItem().getHoverName()))
+                            .getStringWithoutFormatting());
+        }
+
+        if (itemName != null && !itemName.isBlank()) Services.Favorites.toggleFavorite(itemName);
```

Comment:
Remember the brackets :)

Reply:
those sneaky one liners ay :)

Reply:
Hope there are no more formatting errors. I'm new to this modding :D